soccer/src/soccer/strategy/agent/position/robot_factory_position.cpp

```diff
@@ -16,8 +16,63 @@ RobotFactoryPosition::RobotFactoryPosition(int r_id) : Position(r_id) {
 
 std::optional<RobotIntent> RobotFactoryPosition::get_task(WorldState& world_state,
                                                           FieldDimensions& field_dimensions) {
-    // This is where the current_position can be reassigned based on the
-    // PlayState
+    // If keeper, make no changes
+    if (robot_id_ == 0) {
+        return current_position_->get_task(world_state, field_dimensions);
+    }
+
+    // TODO (Rishi and Jack): Make this synchronized across all robots to avoid race conditions
+
+    // Get sorted positions of all friendly robots
+    typedef struct std::pair<int, double> RobotPos;  // (robotId, yPosition)
+
+    std::vector<RobotPos> robots_copy;
+    for (int i = 0; i <= 5; i++) {
```

Comment:
you also shouldn't assume our robots are only 0 through 5; use `kNumShells`, or, better yet, just use  `world_state.our_robots.size()`